app/code/local/Edge/Pages/Model/Observer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+class Edge_Pages_Model_Observer
+{
+    public function cmsField($observer)
+    {
+        $form = $observer->getForm();
+        $fieldset = $form->addFieldset('pagetype_fieldset', array(
+            'legend' => Mage::helper('adminhtml')->__('Page Type'),
+            'class'  => 'fieldset-wide'
+        ));
+
+        $pagestypeCollection = Mage::getModel('pages/page_type')->getCollection();
```

Comment:
for this have a look at 2 functions toOptionArray and toOptionHash

$this->getCollection()->toOptionArray();
$this->getCollection()->toOptionHash();

may do this for you